mem.go

```diff
@@ -80,6 +82,33 @@ func (r RO) EqualString(s string) bool { return r.str() == s }
 // the same bytes.
 func (r RO) EqualBytes(b []byte) bool { return r.str() == string(b) }
 
+var builderPool = sync.Pool{
+	New: func() interface{} {
+		return new(strings.Builder)
+	},
+}
+
+// StringCopy returns m's contents in a newly allocated string.
+func (r RO) StringCopy() string {
+	buf := builderPool.Get().(*strings.Builder)
+	buf.WriteString(r.str())
+	s := buf.String()
+	buf.Reset()
+	builderPool.Put(buf)
```

Comment:
this can be defer after the Get line. People with old Go compilers can pay the cost. That's fine.